### ISSUE TITLE
Some associations in CASE API output are formatted as Data URIs, and may be difficult to consume.

### ISSUE THREAD

**Issue by Lomilar** [opensalt/opensalt#195]
Example:

https://case.georgiastandards.org/ims/case/v1p0/CFAssociations/3fd249e8-416e-11e7-9b21-2c6e3cce03ad

contains a reference (in destinationNodeURI):
data:text/x-ref;src=ASN;type=url,http%3A%2F%2Fasn.desire2learn.com%2Fresources%2FD2769579

Which is a data URI that resolves to http://asn.desire2learn.com/resources/D2769579. While this technically fits the requirements that the reference be a URI, consuming these resources requires the atypical resolution of data URIs (and knowing that you have to resolve data URIs sometimes).

If this isn't just a bug, it could be argued that knowing the resultant format is ASN is useful, but HTTP content negotiation (application/json) and format detection (This looks like rdf+json) plus schema detection (http://www.w3.org/1999/02/22-rdf-syntax-ns#type == http://purl.org/ASN/schema/core/StandardDocument) can serve the same purpose in a much more broadly accepted fashion.

**Comment by roverwolf** [comment]
This was done due to CASE not having any way to add additional information to items or associations.

The general assumption in CASE is that URLs for associations would be other CASE URIs with the exception of exemplars which are assumed to be a URL to a web page containing that exemplar.  Using content negotiation and format detection could work in some cases, but would require a visit to each URL by the system in order to determine what it was (if it could tell).

In this case the association contains a reference to an ASN URL which can provide additional context by pulling data from the URL, but other references such as a UUID (be it a general SIF GUID, AB, EdGate, ...) do not provide any way to get extra information on their own.

My advice at this point would be to ignore URIs that are not understood (and I will open a separate issue about documenting the data URI use in OpenSALT), but I would love to have more discussion on how best to handle things that are not easily done as part of the current CASE specification (such as how to pass and identify an iframable URL for #193).

**Comment by Lomilar** [comment]
The first thing I had to recognize is that in a linked data world, that there's no way that anyone can control what is on the other end of that URL. It could be a 404, a 302, 200, 400, 401, or 500. The responses (OPTIONS requests are nice for this) can have any number of a litany of mimetypes, and in a lot of ways, it's up to the application to decide how to interpret whats on the end of that URL.

Things can change, what was an application/json ASN document could become an HTML page. There's no means for configuration management where we publish how data changes to each other. So, the best that an application can do, IMO, is to interpret what you can, and not crash when presented with the rest.

Linked data is still young. HTTP has no mechanism for negotiating schemas, only formats. There's a long way to go, but don't cut off application profiles that use that data by omitting it if the server doesn't understand it.

That's my current approach.

**Comment by roverwolf** [comment]
I tend to agree with what you wrote, but the way the CASE specification was created it was set up to be consumed by non linked data world (the phrase "linked data" was banned from discussions).
